src/LogViewer/chunk-worker.js

```diff
@@ -127,6 +143,18 @@ const getParagraphClass = (lineNumber, { highlightStart, highlightEnd }) => {
     '';
 };
 
+// Tagged template function
+function escapeHtml(pieces) {
```

Comment:
@helfi92 my pleasure. A key takeaway is that any usage of `.forEach` is a [code smell](https://en.wikipedia.org/wiki/Code_smell), indicating that the iteration relies on manipulating state instead of following a more functional approach. A `.forEach` can be used more functionally with `.map`, `.reduce`, or `.filter`, but not always. Feel free to ask me if you need advice.